src/app/shared/components/grid-list/grid-list.component.ts

```diff
@@ -10,11 +11,14 @@ export class GridListComponent implements OnInit {
     @Input() itemColumnSpan: number = 1;
     @Input() gridGap: number = 0;
 
-    constructor() { }
+    constructor(private clamp: ClampPipe) { }
 
     ngOnInit(): void {
-        this.mobileColumnCount = this.clampColumnValue(this.mobileColumnCount);
-        this.itemColumnSpan = this.clampColumnValue(this.itemColumnSpan);
+        const MAX_COLUMN_COUNT = 12;
```

Comment:
const variable should be defined as below @Input at the top of the class, or event potentially outside the class